src/invoice/signature.rs

```diff
@@ -415,9 +419,32 @@ impl SecretKeyFile {
     }
 }
 
+pub enum LabelMatch<'a> {
+    FullMatch(&'a str),
+    PartialMatch(&'a str),
+    None,
+}
+
+impl<'a> Default for LabelMatch<'a> {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
 impl SecretKeyStorage for SecretKeyFile {
-    fn get_first_matching(&self, role: &SignatureRole) -> Option<&SecretKeyEntry> {
-        self.key.iter().find(|k| k.roles.contains(role))
+    fn get_first_matching(
```

Comment:
Yep, this is much cleaner and has the advantage of being a single function rather than a separate one